src/line/i-chartBuilder.test.js

```diff
@@ -0,0 +1,400 @@
+/**
+ * @jest-environment jsdom
+ */
+
+const fs = require("fs")
+const domTesting = require('@testing-library/dom')
+require('@testing-library/jest-dom')
+const userEvent = require("@testing-library/user-event").default
+
+//Spy on
+const window = require("./window")
+ 
+function initDomFromFiles(htmlPath, jsPath) {
+    window.localStorage.clear()
+    const html = fs.readFileSync(htmlPath, 'utf8')
+    document.open()
+    document.write(html)
+    document.close()
+    jest.isolateModules(function() {
+        require(jsPath)
+    })
+}
+
+/*
+    They should verify the behaviors described above. Note that these behaviors are 
+    duplicated across the three chart builder pages (bar.html, line.html, and scatter.html). 
+    You only need to test these behaviors on one of those pages.
+*/
+
+//ADDING VALUES TO THE CHART BUILDER
+
+test('Pressing "+" adds exactly one empty x and y value box', async function(){
+    initDomFromFiles(`${__dirname}/line.html`, `${__dirname}/line.js`)
+
+    //get + button for adding  new x and y cells.
+    const buttons = domTesting.queryAllByRole(document, "button")
+    const addButton = buttons[3];
+    
+    //make sure the correct button is selected
+    expect(addButton).toHaveTextContent("+")
```

Comment:
I'm not in love with this method for getting buttons. I see how this verifies that the test is clicking the button it expects to, but it's still a brittle solution. If the ordering of the buttons changes for any reason, this assertion would cause the test to fail even if the SUT functions properly.

I think this should be changed throughout the integration tests to instead use a `.getByText()` query, maybe using the optional parameters for that function to ensure that the selected element is a button if you'd like to ensure that.